jetson/nucleo_bridge/README.md

```diff
@@ -0,0 +1,104 @@
+Code to control the RA/SA, using Nucleo F303REs
+===============================================
+### About
+
+nucleo_bridge is responsible for connecting the rover network's LCM interface with the RA/SA system's Nucleo controller i2c interface
+
+main.cpp calls init() on the static LCMHandler class \
+main.cpp calls init() on the static ControllerMap class \
+main.cpp calls init() on the static I2C class \
+main.cpp creates two threads to run an outgoing function and an incoming function
+The outgoing function calls on the LCMHandler's handle_outgoing() function every millisecond
+The incoming function calls on the LCMHandler's handle_incoming() function continuously
+
+The ControllerMap class creates a hash table of virtual Controller objects from the config file located at "mrover-workspace/config_nucleo_bridge/controller_config.json".These virtual Controllers are used to contact the physical controller on the rover, across both RA/SA configurations.
+
+The virtual Controller class is defined in Controller.h.\
+Virtual Controllers store information about various controller-specific parameters (such as encoder cpr)\
+The virtual Controller class also has functions representing the possible transactions that can be had with the physical controller. \
+The virtual Controller will not attempt to communicate with its physical controller unless "activated" by an appropriate LCM message relayed by LCMHandler.h
+(e.g. A virtual RA Controller will never attempt to communicate with its physical RA controller unless an RA-related LCM message is sent. This is to prevent multiple virtual Controller objects from trying to contact the same physical Controller object.)
+
+LCMHandler.h is responsible for handling incoming and outgoing lcm messages. \
+Incoming lcm messages will trigger functions which call the functions on the appropriate virtual Controllers. \
+Outgoing lcm messages are triggered by a clock, which query the functions on the appropriate virtual Controllers for data.
+
+I2C.h is responsible for translating communications by virtual Controllers into i2c transactions understood by the linux drivers.
+
+### LCM Channels
+#### RA Open Loop \[Subscriber\] "/ra_openloop_cmd"
+Message: [RAOpenLoopCmd.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/RAOpenLoopCmd.lcm) \
+Publisher: onboard/teleop \

```

Comment:
update all of these from onboard to jetson

Reply:
fixed